jbmc/regression/jbmc-concurrency/synchronized-method-illegal-state/test.desc

```diff
@@ -0,0 +1,13 @@
+KNOWNBUG
+Sync.class
+--cp ../../../src/java_bytecode/library/core-models.jar:. --lazy-methods --java-threading
+
+^VERIFICATION SUCCESSFUL$
+^EXIT=0$
+^SIGNAL=0$
+--
+^warning: ignoring
+--
+Checks all possible paths to ensure a synchronized method does not end in an illegal monitor state.
+
+This is currently not working as explicit throws have been removed from the underlying model due to performance considerations.
```

Comment:
I'm not sure this is a known bug. The code in the `.java` file should be supported without problem by this PR. What this phrase refers to is the lack of throwing a runtime exception in our model of `monitorenter/exit` *when there are reasons to throw such exceptions* (i.e, synchronizing on a null object or calling `monitorexit` when the counter is 0), but no such conditions are triggered in this test. Therefore this should work.

Am I missing something?